db/sg_replicate_cfg.go

```diff
@@ -1217,6 +1219,11 @@ func (c *SGRCluster) RebalanceReplications() {
 
 	// If there aren't any nodes available, there's nothing more to be done
 	if len(c.Nodes) == 0 {
+		if len(unassignedReplications) > 0 {
+			for _, v := range unassignedReplications {
+				base.WarnfCtx(c.loggingCtx, "Replication %s does not have an assigned node.", v.ID)
+			}
+		}
```

Comment:
nit: we can range over an empty (or even nil) slice/map to avoid the length check

```suggestion
		for _, v := range unassignedReplications {
			base.WarnfCtx(c.loggingCtx, "Replication %s does not have an assigned node.", v.ID)
		}
```